code/game/gamemodes/changeling/powers/hivemind.dm

```diff
@@ -71,9 +71,10 @@ var/list/datum/dna/hivemind_bank = list()
 /obj/effect/proc_holder/changeling/hivemind_download/sting_action(mob/user)
 	var/datum/changeling/changeling = user.mind.changeling
 	var/list/names = list()
-	for(var/datum/changelingprofile/prof in hivemind_bank)
-		if(!(prof in changeling.stored_profiles))
-			names[prof.name] = prof
+	for(var/datum/changelingprofile/hivemind_prof in hivemind_bank)
+		for(var/datum/changelingprofile/stored_prof in changeling.stored_profiles)
+			if(stored_prof.name != hivemind_prof.name)
+				names[hivemind_prof.name] = hivemind_prof
```

Comment:
This is kinda dumb I think because you'll still populate names. Even if you selected someone you've already absorbed it's handled down below